### PR TITLE
fix: network upgrade with wraps and configuration files updates

### LINKED ISSUE #3923: Solo consensus network upgrade can fail copying --local-build-path
## Summary

`solo consensus network upgrade` can fail during the local build copy step when upgrading a 3-node Kind-based consensus network from `v0.72.0-rc.2` to a local `0.73.0-SNAPSHOT` build.

Solo starts the upgrade, freezes the network, updates chart/config state, and then tries to copy the local build into the restarted consensus pods. In this run, the copy raced with pod lifecycle changes and Solo failed with:

`Error in upgrade network: Error in copying local build to node: ... kubectl cp ... failed with code 1: error: cannot exec into a container in a completed pod; current phase is Failed`

## Environment

- Solo: `0.67.0`
- Kubernetes: Kind `kind-solo-tss-upgrade`
- Baseline release: `v0.72.0-rc.2`
- Upgrade release: `v0.73.0-SNAPSHOT` (with tss enabled)
- Local build path: `hedera-node/data`
- 3 consensus nodes: `node1,node2,node3`

### Describe the expected behavior

Solo should wait for the replacement pod/container to become valid before attempting the local-build copy, or retry with sufficient backoff so that transient pod restarts do not fail the whole upgrade.

## Reproduction

I reproduced this from our wrapper script, but the relevant upgrade invocation is:

```bash
solo consensus network upgrade \
  --deployment solo-tss-upgrade \
  --node-aliases node1,node2,node3 \
  --upgrade-version v0.73.0-rc.1 \
  --local-build-path /Users/ivanbankov/Workspace/hedera-services/hedera-node/data \
  --application-properties /Users/ivanbankov/Workspace/hedera-services/hedera-node/test-clients/scripts/solo/e2e-block-stream-cutover/resources/0.73/application.properties \
  --application-env /Users/ivanbankov/Workspace/hedera-services/hedera-node/test-clients/scripts/solo/e2e-block-stream-cutover/resources/0.73/application.env \
  --wraps-key-path /Users/ivanbankov/.solo/cache/wraps-v0.2.0 \
  --force

### Additional Context

[solo-copy-issue-2026-04-07.tar.gz](https://github.com/user-attachments/files/26530313/solo-copy-issue-2026-04-07.tar.gz)

### PR DESCRIPTION
**Pull request by Ivo-Yankov**
## Description

This PR fixes an issue that can occur when performing `solo consensus network upgrade` on a deployment with `wraps` and `tss` enabled. The issue is that when configuration files are passed, we perform a `helm chart upgrade`, which causes the network pods to terminate and restart after a short delay. However, before the pods have fully restarted, solo attempts to upload the wraps file to the root container, resulting in an error.

### Related Issues

* Closes #3923
* Related to #

### Pull request (PR) checklist

* \[ ] This PR added tests (unit, integration, and/or end-to-end)
* \[ ] This PR updated documentation
* \[ ] This PR added no TODOs or commented out code
* \[ ] This PR has no breaking changes
* \[ ] Any technical debt has been documented as a separate issue and linked to this PR
* \[ ] Any `package.json` changes have been explained to and approved by a repository manager
* \[ ] All related issues have been linked to this PR
* \[ ] All changes in this PR are included in the description
* \[ ] When this PR merges the commits will be squashed and the title will be used as the commit message, the 'commit message guidelines' below have been followed

### Testing

* \[ ] This PR added unit tests
* \[ ] This PR added integration/end-to-end tests
* \[ ] These changes required manual testing that is documented below
* \[ ] Anything not tested is documented

The following manual testing was done:

* TBD

The following was not tested:

* TBD

<details>
<summary>
Commit message guidelines
</summary>
We use 'Conventional Commits' to ensure that our commit messages are easy to read, follow a consistent format, and for automated release note generation. Please follow the guidelines below when writing your commit messages:

1. BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with MAJOR in Semantic Versioning). A BREAKING CHANGE can be part of commits of any type.  NOTE: currently breaking changes will only bump the MAJOR version.
2. The title is prefixed with one of the following:

| Prefix    | Description                                         | Semantic Version Update | Captured in Release Notes |
|-----------|-----------------------------------------------------|-------------------------|---------------------------|
| feat:     | a new feature                                       | MINOR                   | Yes                       |
| fix:      | a bug fix                                           | PATCH                   | Yes                       |
| perf:     | performance                                         | PATCH                   | Yes                       |
| refactor: | code change that isn't feature or fix               | none                    | No                        |
| test:     | adding missing tests                                | none                    | No                        |
| docs:     | changes to documentation                            | none                    | Yes                        |
| build:    | changes to build process                            | none                    | No                        |
| ci:       | changes to CI configuration                         | none                    | No                        |
| style:    | formatting, missing semi-colons, etc                | none                    | No                        |
| chore:    | updating grunt tasks etc; no production code change | none                    | No                        |

</details>

### CODE REVIEW

File: src/commands/node/tasks.ts [hunk 1]
```diff
@@ -3163,7 +3180,25 @@ export class NodeCommandTasks {
...
+            try {
+              if (await rootContainer.execContainer(`test -d "${targetWrapsPath}"`)) {
+                found = true;
+                break;
+              }
+            } catch {
+              this.logger.info(
+                `Attempt ${attempt}/${attempts}: WRAPs directory not found in node ${consensusNode.name}. Retrying...`,
+              );
+              await sleep(Duration.ofSeconds(2));
```
Comment (JeffreyDallas) [inline src/commands/node/tasks.ts]: move to constant.ts, similar to other WAIT ATTEMPT

File: src/commands/node/tasks.ts [hunk 2]
```diff
@@ -3163,7 +3180,25 @@ export class NodeCommandTasks {
 
           const targetWrapsPath: string = `${constants.HEDERA_HAPI_PATH}/${wraps.directoryName}`;
 
-          if (await rootContainer.execContainer(`test -d "${targetWrapsPath}"`)) {
+          const attempts: number = 10;
```
Comment (JeffreyDallas) [inline src/commands/node/tasks.ts]: move to constant.ts, similar to other WAIT ATTEMPT

**Review by JeffreyDallas — COMMENTED** [review]
Since you are working on network ready, may be you can support this issue too

https://github.com/hiero-ledger/solo/issues/4016

**Review by jan-milenkov — DISMISSED** [review]

File: docs/site/content/en/docs/env.md [hunk 3]
```diff
@@ -12,111 +12,113 @@ User can configure the following environment variables to customize the behavior
...
+| `JAVA_FLIGHT_RECORDER_CONFIGURATION`               | Path to a JFR configuration file to be uploaded to the consensus nodes. Can be used as a substitute for the `--jfr-config` flag | `` (empty)                                                                                         |
+| `TRIGGER_STAKE_WEIGHT_CALCULATE_WAIT_SECONDS`      | The number of seconds to wait after triggering stake weight calculation.                                                        | `60`                                                                                               |
+| `CHECK_WRAPS_DIRECTORY_MAX_ATTEMPTS`               | The maximum number of attempts to check for the existance of the wraps directory.                                               | `10`                                                                                               |
+| `CHECK_WRAPS_DIRECTORY_BACKOFF_MS`                 | The backoff period for retrying to check for the existance of the wraps directory.                                              | `2000`                                                                                             |
```
Comment (JeffreyDallas) [inline docs/site/content/en/docs/env.md:121]: why such a huge format change of the whole section

**Review by JeffreyDallas — DISMISSED** [review]

**Review by jan-milenkov — APPROVED** [review]

File: docs/site/content/en/docs/env.md (hunk 3 above)
Comment (Ivo-Yankov) [reply docs/site/content/en/docs/env.md:121]: I added 2 new entries to the table and reformatted it. I guess that changed the width of some columns in the whole table.

**Review by jeromy-cannon — APPROVED** [review]
